### ISSUE TITLE
regression in 0.13.10-SNAPSHOT maven style publishing

### ISSUE THREAD

**Issue by fommil** [sbt/sbt#2347]
I don't get this in 0.13.9.

We recently added a new `Configuration` to the `ensime-sbt` plugin https://github.com/ensime/ensime-sbt/blob/5f18b5aac2b6bd30dc829414e782e6ce3461d5c2/src/main/scala/EnsimePlugin.scala#L98

It is marked as `hide` and shouldn't appear anywhere in the project's `pom`.

However, when running a really simple project https://github.com/fommil/sbt-big-project/tree/master/src/sbt-test/sbt-big-project/simple with the new plugin, I see this

```
[warn] Found intransitive dependency (org.scala-lang:scala-compiler:2.10.6:ensime-internal) while publishMavenStyle is true, but Maven repositories
[warn]   do not support intransitive dependencies. Use exclusions instead so transitive dependencies
[warn]   will be correctly excluded in dependent projects.
[warn]            
```

Basically, this shouldn't even be a consideration so the warning shouldn't appear.

**Comment by eed3si9n** [comment]
Thanks for the report. It looks like a regression on #2345/#2127.

**Comment by dwijnand** [comment]
Echoing my thanks for testing with the SNAPSHOT and opening this ticket.

**Comment by fommil** [comment]
I'm now wondering if this is a bug. Now that I understand why the error appears I can see it would be useful. The thing is that I want to suppress these warnings in my case because I'm not publishing them.

**Comment by eed3si9n** [comment]
See #2407. It's now going to warn on Compile and Runtime only by default.
